craft_providers/lxd/lxd_instance.py

```diff
@@ -0,0 +1,494 @@
+        """
+        disks = self._get_disk_devices()
+
+        return any(
+            disk["path"] == target.as_posix()
+            and disk["source"] == host_source.as_posix()
+            for _, disk in disks.items()
+        )
+
+    def is_running(self) -> bool:
+        """Check if instance is running.
+
+        :returns: True if instance is running.
+
+        :raises LXDError: On unexpected error.
+        """
+        state = self._get_state()
+        if state is None:
+            raise LXDError(brief=f"Instance {self.name!r} does not exist.")
+
+        return state.get("status") == "Running"
+
+    def launch(
+        self,
+        *,
+        image: str,
+        image_remote: str,
+        map_user_uid: bool = False,
+        ephemeral: bool = False,
+    ) -> None:
+        """Launch instance.
+
+        :param image: Image name to launch.
+        :param image_remote: Image remote name.
+        :param uid: Host user ID to map to instance root.
+        :param ephemeral: Flag to enable ephemeral instance.
+
+        :raises LXDError: On unexpected error.
+        """
+        config_keys = dict()
+
+        if map_user_uid:
+            uid = os.getuid()
+            config_keys["raw.idmap"] = f"both {uid!s} 0"
+
+        if self._host_supports_mknod():
+            config_keys["security.syscalls.intercept.mknod"] = "true"
+
+        self.lxc.launch(
+            config_keys=config_keys,
+            ephemeral=ephemeral,
+            instance_name=self.name,
+            image=image,
+            image_remote=image_remote,
+            project=self.project,
+            remote=self.remote,
+        )
+
+    def mount(
+        self,
+        *,
+        host_source: pathlib.Path,
+        target: pathlib.Path,
+        device_name: Optional[str] = None,
+    ) -> None:
+        """Mount host source directory to target mount point.
+
+        Checks first to see if already mounted. If no device name is
+        given, it will be generated with the format:
+         disk_<target.as_posix().replace("/", "_")>
+
+        For example:  /mnt/foo will be named "disk__mnt_foo"
+
+        :param host_source: Host path to mount.
+        :param target: Instance path to mount to.
+        :param device_name: Name for disk device.
+
+        :raises LXDError: On unexpected error.
+        """
+        if device_name is None:
+            device_name = "disk_" + target.as_posix().replace("/", "_")
+
+        if self.is_mounted(host_source=host_source, target=target):
+            return
+
+        self.lxc.config_device_add_disk(
+            instance_name=self.name,
+            source=host_source,
+            path=target,
+            device=device_name,
+            project=self.project,
+            remote=self.remote,
+        )
+
+    def _host_supports_mknod(self) -> bool:
+        """Check if host supports mknod in container.
+
+        See: https://actions.linuxcontainers.org/lxd/docs/master/syscall-interception
```

Comment:
Is the `actions` name correct here? It doesn't resolve for me, and the context exists at `linuxcontainers.org`.

Reply:
Will fix

Reply:
Fixed